src/configure/configurers/provisioningConfigurer.ts

```diff
@@ -0,0 +1,267 @@
+                    } else {
+                        throw new Error("Failed to configure pipeline");
+                    }
+                } catch (error) {
+                    telemetryHelper.logError(Layer, TracePoints.RemotePipelineConfiguringFailed, error);
+                    vscode.window.showErrorMessage(utils.format(Messages.ConfiguringPipelineFailed, error.message));
+                    return null;
+                }
+            });
+        } else {
+                telemetryHelper.setTelemetry(TelemetryKeys.PipelineDiscarded, 'true');
+                throw new UserCancelledError(Messages.operationCancelled);
+        }
+
+        if ( provisioningServiceResponse != undefined) {
+            this.setQueuedPipelineUrl(provisioningServiceResponse, inputs);
+        } else {
+            throw new Error("Failed to configure provisoining pipeline");
+        }
+    }
+
+    public async showPipelineFiles(): Promise<void> {
+        this.filesToCommit.forEach(async (file) => {
+            await this.localGitRepoHelper.addContentToFile(file.content, file.absPath);
+            await vscode.window.showTextDocument(vscode.Uri.file(file.absPath));
+        });
+    }
+
+    public setQueuedPipelineUrl(provisioningConfiguration: ProvisioningConfiguration, inputs: WizardInputs){
+      const commitId = (provisioningConfiguration.result.pipelineConfiguration as CompletePipelineConfiguration).commitId;
+      this.queuedPipelineUrl = `https://github.com/${inputs.sourceRepository.repositoryId}/commit/${commitId}/checks`;
+    }
+
+    public async  getFileToCommit(draftPipelineConfiguration: DraftPipelineConfiguration): Promise<void> {
+        let destination: string;
+        for (const file of draftPipelineConfiguration.files ) {
+            destination = await this.getPathToFile(Path.basename(file.path), Path.dirname(file.path));
+            const decodedData = new Buffer(file.content, 'base64').toString('utf-8');
+            this.filesToCommit.push({absPath: destination, content: decodedData, path: file.path} as DraftFile);
+        }
+    }
+
+    public async getPathToFile( fileName: string, directory: string) {
+        const dirList = directory.split("/"); // Hardcoded as provisioning service is running on linux and we cannot use Path.sep as it is machine dependent
+        let directoryPath: string = "";
+        directoryPath = await this.localGitRepoHelper.getGitRootDirectory();
+        dirList.forEach((dir) => {
+            try {
+                directoryPath = Path.join(directoryPath, dir);
+                // tslint:disable-next-line:non-literal-fs-path
+                if (!fs.existsSync(directoryPath)) {
+                    // tslint:disable-next-line:non-literal-fs-path
+                    fs.mkdirSync(directoryPath);
+                }
+            }
+            catch (error) {
```

Comment:
you don't need to catch error just to throw it again. If you don't write a catch block, the error will be automatically thrown to parent

Reply:
kept try and cath and added logging in catch block